### ISSUE TITLE
Check whether running in simulation mode

### ISSUE THREAD

**Issue by letmaik** [openenclave/openenclave#3173]
I need to check from within an enclave whether it is running in simulation mode or not. I couldn't find an API for that. Am I missing something?

**Comment by jhand2** [comment]
I believe you should be able to do:

```
oe_enclave_t *enclave = oe_get_enclave();
bool is_simulation = enclave->simulate;
```

**Comment by anakrish** [comment]
I don't think we publish oe_enclave struct; the user doesn't have access to it.

I think it is good to add the following APIs:
```
bool is_simulation_mode();
bool is_debugger_attached(); // This is part of debugger contract; but we haven't implemented this yet.
```

**Comment by jhand2** [comment]
@anakrish in that case, should we consider moving oe_get_enclave in internal? Or at least consider making it's type an opaque handle. It is published in `enclave.h`. `oe_enclave_t` is in `bits/types.h` which is probably ok considering it's a bits header.

**Comment by letmaik** [comment]
@jhand2 `oe_enclave_t` is *not* defined as full type in the public `bits/types.h`. Is there a hacky work-around while this feature is being implemented?

**Comment by anakrish** [comment]
@jhand2 IIRC, `oe_get_enclave` was added in [v0.40](https://github.com/openenclave/openenclave/blob/95ac743e6c60775943e5745469c9cca957591a22/CHANGELOG.md#v040---2018-10-08-deprecated) based on request from users to provide some kind of handle that the enclave can send to the host. Deprecating `oe_get_enclave` would have to be done, if needed, in a phased manner.

**Comment by anakrish** [comment]
@letmaik 
RE: workarounds
One approach that doesn't require access to internal data structures would be to just add an EDL ecall `enc_set_simulate()` and call it from the host.
Other approaches would require having definitions of `td_t` which has a simulate field.
